app/services/payment_requests/create_service.rb

```diff
@@ -10,19 +10,34 @@ def initialize(organization:, params:)
     end
 
     def call
-      return result.not_found_failure!(resource: "customer") unless customer
+      # TODO: Return error unless premium license
 
-      ActiveRecord::Base.transaction do
-        # TODO: Create payable group for the overdue invoices
+      return result.not_found_failure!(resource: "customer") unless customer
+      return result.not_found_failure!(resource: "invoice") if invoices.empty?
 
-        # TODO: Create payment request for the payable group
+      # TODO: Return error if invoices are not overdue
 
-        # TODO: Send payment_request.created webhook
+      ActiveRecord::Base.transaction do
+        # NOTE: Create payable group for the overdue invoices
+        payable_group = customer.payable_groups.create!(organization:)
+        invoices.each { |i| i.update!(payable_group:) }
```

Comment:
would you expect any side effect or validation to happen here. if not we could use `update_all` to perform a unique SQL statement in order to update all the invoices at once.

what do you think?

Reply:
Yeah - I was hesitant but agreed let's update all the invoices at once 👍